### ISSUE TITLE
[Feat Req] Add a picker that would output a formatted citation

### ISSUE THREAD

**Issue by lambtho12** [nvim-telescope/telescope-bibtex.nvim#30]
For people using telescope-bibtex for note taking (in markdown for instance), it would be nice to be able to format the citation directly (following some of the most common conventions, such as APA, Chicago, etc). 

That way, it would be extremely simple to generate notes of that sort:
```markdown
## My note
Lorem ipsum dolor sit amet, consectetur adipiscing elit.[@citekey]. 

## References
[@citekey]: Name F., 2022, "My awesome source", Journal of Neovim, Paris, France
``` 

Some configuration variables could be left to the user, such as:
- the choice of the convention: APA, Chicago, ...
- the output format: `only_citation`, `[@citekey]: citation`, ...

**Comment by noahares** [comment]
So, you whish to produce the text unter References by parsing more details from the entry?
If so, this should be relatively easy to acomplish. I will have a look at it (hopefully this weekend) after we merge the changes to the pickers from #31

**Comment by noahares** [comment]
I love the idea of Telekasten. Currently I still use vimwiki because I got exams around the corner and do not want to move all my notes, but definitly will check it out some time. 
Telescope-bibtex should not need to do extra formating of entries if I understood this correctly? It just extracts the fields and display those specified by the convention or the user and maybe adds extra characters like (year) or "title". 
The plan would be to support the formats you mentioned (APA, Chicago and maybe one more) out of the box and then provide a user specific template.

**Comment by lambtho12** [comment]
> Telescope-bibtex should not need to do extra formating of entries if I understood this correctly? It just extracts the fields and display those specified by the convention or the user and maybe adds extra characters like (year) or "title".

Yes. Scape the bibtex entry for the selected citation, concatenate all information scraped all in a long string accordingly with the style chosen in the options and then paste that.


> The plan would be to support the formats you mentioned (APA, Chicago and maybe one more) out of the box and then provide a user specific template.

The best future-proof solution would be to implement [CSL (Citation Style Language)](https://citationstyles.org/developers/), which is the "standard" with all citation styles. This is basically what Zotero, Mendelay, etc use. It seems that a lua implementation exists ([see here](https://giters.com/zepinglee/citeproc-lua)) but I have not tested it. That would support _all_ styles out-of-the-box. But this may also be a bit trickier to implement. 
In the meantime, I think that implementing 2-3 very common styles and add the possibility to have user-defined template would likely satisfy most use-cases.

**Comment by noahares** [comment]
> The best future-proof solution would be to implement CSL (Citation Style Language), which is the "standard" with all citation styles. This is basically what Zotero, Mendelay, etc use. It seems that a lua implementation exists (see here) but I have not tested it. That would support all styles out-of-the-box. But this may also be a bit trickier to implement.
In the meantime, I think that implementing 2-3 very common styles and add the possibility to have user-defined template would likely satisfy most use-cases.

This sounds like a good plan. Having the common styles implemented to have this feature for people to use it and then try to make CSL work.

> As you are open to the idea, I will try to find some time this week to do a basic implementation of the feature.

That would be great. I will do what I can to help.

**Comment by lambtho12** [comment]
Solved by #33
